File: unidbg-api/src/main/java/com/vortexdbg/hook/HookListener.java
package com.vortexdbg.hook;

import com.vortexdbg.memory.SvcMemory;

public interface HookListener {

    int EACH_BIND = -1;
    int WEAK_BIND = -2;
    int FIXUP_BIND = -3;

    /**
     * 返回0表示没有hook，否则返回hook以后的调用地址
     */
    long hook(SvcMemory svcMemory, String libraryName, String symbolName, long old);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/hook/HookListener.kt
package com.vortexdbg.hook

import com.vortexdbg.memory.SvcMemory

interface HookListener {

    /**
     * Decide whether to redirect a symbol binding.
     *
     * @return 0 to leave the binding untouched, otherwise the address calls
     *         to this symbol should be redirected to.
     */
    fun hook(svcMemory: SvcMemory, libraryName: String, symbolName: String, old: Long): Long

    companion object {
        const val EACH_BIND = -1
        const val WEAK_BIND = -2
        const val FIXUP_BIND = -3
    }

}
